api/src/org/labkey/api/exp/OntologyManager.java

```
@@ -2078,6 +2078,101 @@ public static PropertyDescriptor getPropertyDescriptor(String propertyURI, Conta
         return propDescCache.get(key);
     }
 
+
+    public static List<PropertyDescriptor> getPropertyDescriptors(
+            Container c, User user,
+            @Nullable Set<Integer> domainIds,
+            @Nullable Set<String> domainKinds,
+            @Nullable String searchTerm,
+            @Nullable SimpleFilter propertyFilter,
+            @Nullable String sortColumn,
+            @Nullable Integer maxRows, @Nullable Long offset)
+    {
+        final FieldKey propertyIdKey = FieldKey.fromParts("propertyId");
+
+        // To filter by domain kind, we query the exp.DomainProperty table and filter by domainId.
+        // To construct a PropertyDescriptor, we will need to traverse the lookup to exp.PropertyDescriptor and select all of it's columns.
+        List<FieldKey> fields = new ArrayList<>();
+        fields.add(FieldKey.fromParts("domainId"));
+        for (ColumnInfo col : getTinfoPropertyDescriptor().getColumns())
+        {
+            fields.add(new FieldKey(propertyIdKey, col.getName()));
+        }
+        var colMap = QueryService.get().getColumns(getTinfoPropertyDomain(), fields);
+
+        var filter = new SimpleFilter();
+        if (propertyFilter != null)
+        {
+            filter.addAllClauses(propertyFilter);
+        }
+
+        filter.addCondition(new FieldKey(propertyIdKey, "container"), c.getId());
+        if (domainIds != null && !domainIds.isEmpty())

```

Should we throw an exception if you pass both domainIds and domainKinds as arguments? domainIds win and causes us to ignore domainKinds

---

I went back and forth on this.  I changed this to allow both domainIds and domainKinds to be specified.  I also added validation of the other form parameters